### PR TITLE
test: rework 6794 test to generate broken data files

### LINKED ISSUE #8701: Regenerate `test/box-luatest/gh_6794_data`
The [directory](https://github.com/tarantool/tarantool/tree/cde911d0e0a65a17d39b5ac60bae2bc2fe63bead/test/box-luatest/gh_6794_data) contains snap and xlog files that are used to start a test instance. The files use an old schema (the schema was upgraded after they were generated). In the scope of #7149 we disable any DDL with an old schema version. The problem is luatest needs to grant permissions (which is a DDL operation) to run tests (it does it in the test server instance source file [here](https://github.com/tarantool/luatest/blob/89da427f8bb3bb66e01d2a7b5a9370d0428d8c52/luatest/server_instance.lua#L47)). As a result, the [test](https://github.com/tarantool/tarantool/blob/cde911d0e0a65a17d39b5ac60bae2bc2fe63bead/test/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua) fails while trying to start a test server instance.

To fix this issue, we need to regenerate the data files with `box.schema.user.grant('guest', 'super')`. Unfortunately, there's no script that could be used for this (like [this one](https://github.com/tarantool/tarantool/blob/cde911d0e0a65a17d39b5ac60bae2bc2fe63bead/test/box-luatest/upgrade/2.10.4/gen.lua), for example) so they have to be regenerated manually.

It'd be nice if in the scope of this ticket we'd also add a script that would regenerate the data files automatically.

### PR DESCRIPTION
**Pull request by sergepetrenko**
Instead of relying on fixed xlog / snap pair in gh_6794_data, let's better generate the desired files right in the test. This way we won't face the problem of the files getting out of date. For example, when schema version changes.

Remove the now unneeded gh_6794_data files and re-enable the gh_6794_recover_nonmatching_xlogs test.

Closes #8701

NO_DOC=test
NO_CHANGELOG=test

### CODE REVIEW

File: test/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua [hunk 1]
```diff
@@ -4,9 +4,54 @@ local fio = require('fio')
...
+    cg.xlog:stop()
+    cg.xlog_path = fio.pathjoin(cg.xlog.workdir, '00000000000000000000.xlog')
+    t.assert_equals(#fio.glob(cg.xlog_path), 1, 'Xlog is created')
+
+    cg.tempdir = fio.tempdir()
+    fio.copyfile(cg.snap_path, cg.tempdir)
+    fio.copyfile(cg.xlog_path, cg.tempdir)
+end)
+
+g.after_all(function(cg)
```
Comment (grafin) [inline test/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua:46]: Just to be sure (didn't find it in luatest doc): after_all does run always, even if one or more tests fail?

**Review by grafin — APPROVED** [review]
Thanks for the patch! Just one question from me)

File: test/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua (hunk 1 above)
Comment (sergepetrenko) [reply test/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua:46]: Yep, it does. I checked like this:
<details><summary>Applied the difff:</summary>

```diff
diff --git a/test/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua b/test/box-luatest/gh_6794_>
index 156d8264f..c71830593 100644
--- a/test/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua
+++ b/test/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua
@@ -47,6 +47,8 @@ g.after_all(function(cg)
     cg.snap:drop()
     cg.xlog:drop()
     fio.rmtree(cg.tempdir)
+    require("log").info("Run after_all")
+    t.assert(false, 'in after all')
 end)
 
 g.before_test('test_panic_without_force_recovery', function()
@@ -78,6 +80,7 @@ g.test_panic_without_force_recovery = function()
         local filename = fio.pathjoin(g.server.workdir, g.server.alias..'.log')
         t.assert(g.server:grep_log(msg, nil, {filename = filename}))
     end)
+    t.assert(false, 'in test')
 end
 
 g.test_ignore_with_force_recovery = function()
@@ -93,4 +96,5 @@ g.test_ignore_with_force_recovery = function()
         t.assert_equals(box.info.status, 'running', 'The server is recovered')
         t.assert(not box.info.ro, 'Server is writable')
     end)
+    t.assert(false, 'in test')
 end
```
</details>

And ran the test:
```
[001] not ok 1	box-luatest.gh_6794_recover_nonmatching_xlogs.test_panic_without_force_recovery
[001] #   ...t/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua:83: in test
[001] #   expected: a value evaluating to true, actual: false
[001] #   stack traceback:
[001] #   	...t/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua:83: in function 'box-luatest.gh_6794_recover_nonmatching_xlogs.test_panic_without_force_recovery'
[001] #   	...
[001] #   	[C]: in function 'xpcall'
[001] #   artifacts:
[001] #   	snapshot -> /tmp/t/001_box-luatest/artifacts/snapshot-7-lkPQAAzFL4
[001] #   	xlog -> /tmp/t/001_box-luatest/artifacts/xlog-xrh0wMBmKACo
[001] #   	master-test_panic -> /tmp/t/001_box-luatest/artifacts/master-test_panic-yXBpjv5Jxbyv
[001] not ok 2	box-luatest.gh_6794_recover_nonmatching_xlogs.test_ignore_with_force_recovery
[001] #   ...t/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua:99: in test
[001] #   expected: a value evaluating to true, actual: false
[001] #   stack traceback:
[001] #   	...t/box-luatest/gh_6794_recover_nonmatching_xlogs_test.lua:99: in function 'box-luatest.gh_6794_recover_nonmatching_xlogs.test_ignore_with_force_recovery'
[001] #   	...
[001] #   	[C]: in function 'xpcall'
[001] #   artifacts:
[001] #   	master-test_ignore -> /tmp/t/001_box-luatest/artifacts/master-test_ignore-KKQL3-XGVb1n
[001] # Ran 2 tests in 0.771 seconds, 0 succeeded, 2 failed
[001] 
[001] [test-run server "luatest_server"] Last 15 lines of the log file /tmp/t/001_box-luatest/gh_6794_recover_nonmatching_xlogs_test.log:
[001] Run after_all
```
The test logs contain `Run after_all` at `/tmp/t/001_box-luatest/gh_6794_recover_nonmatching_xlogs_test.log`

**Review by Gerold103 — APPROVED** [review]
Thanks for the patch!

**Pull request merged**